src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/SnsService.kt

```diff
@@ -0,0 +1,132 @@
+package uk.gov.justice.digital.hmpps.visitscheduler.service
+
+import com.amazonaws.services.sns.model.MessageAttributeValue
+import com.amazonaws.services.sns.model.PublishRequest
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.visitscheduler.dto.VisitDto
+import uk.gov.justice.digital.hmpps.visitscheduler.service.SnsService.Companion.EVENT_ZONE_ID
+import uk.gov.justice.hmpps.sqs.HmppsQueueService
+import java.time.Instant
+import java.time.ZoneId
+import java.time.format.DateTimeFormatter
+import java.util.function.Supplier
+
+/**
+ * SQS Reference
+ * https://github.com/ministryofjustice/hmpps-spring-boot-sqs
+ */
+
+@Service
+class SnsService(hmppsQueueService: HmppsQueueService, private val objectMapper: ObjectMapper) {
+
+  private val domaineventsTopic by lazy { hmppsQueueService.findByTopicId(TOPIC_ID) ?: throw RuntimeException("Topic with name $TOPIC_ID doesn't exist") }
+  private val domaineventsTopicClient by lazy { domaineventsTopic.snsClient }
+
+  fun sendVisitBookedEvent(visit: VisitDto) {
+    log.debug("Sending visit booked event")
+    publishToDomainEventsTopic(
+      HMPPSDomainEvent(
+        eventType = EVENT_PRISON_VISIT_BOOKED,
+        version = EVENT_PRISON_VISIT_VERSION,
+        description = EVENT_PRISON_VISIT_BOOKED_DESC,
+        occurredAt = Instant.now().toString(),
+        prisonerId = visit.prisonerId,
+        additionalInformation = AdditionalInformation(
+          reference = visit.reference,
+        )
+      )
+    )
+  }
+
+  fun sendVisitCancelledEvent(visit: VisitDto) {
+    log.debug("Sending visit cancelled event")
+    publishToDomainEventsTopic(
+      HMPPSDomainEvent(
+        eventType = EVENT_PRISON_VISIT_CANCELLED,
+        version = EVENT_PRISON_VISIT_VERSION,
+        description = EVENT_PRISON_VISIT_CANCELLED_DESC,
+        occurredAt = Instant.now().toString(),
```

Comment:
```
        occurredAt = LocalDateTime.now().toString()
Or
        occurredAt = LocalDate.now().toString()
```

Reply:
Yeah, although it works and matches the status quo, i'm not overly happy with this code.